app/models/manageiq/providers/kubernetes/container_manager/refresh_parser.rb

```diff
@@ -128,12 +128,57 @@ def process_collection_item(item, key)
       new_result
     end
 
+    def find_host_instance(provider_id)
+      providers = {
+        :gce => [ManageIQ::Providers::Google::CloudManager, :name],
+        :aws => [ManageIQ::Providers::Amazon::CloudManager, :uid_ems],
+      }
+      instance_uri = URI(provider_id)
+      prov, name_field = providers[instance_uri.scheme.to_sym]
+      return nil unless prov
+      project_name = instance_uri.host
+      _, az, instance_name = instance_uri.path.split('/')
+
+      dits = {name_field => instance_name}
+      dits[:availability_zone_id] =
+        AvailabilityZone.find_by(:name => az).try(:id) if az != ""
+
+      if project_name
+        prov = prov.find_by_project(project_name)
+        return nil unless prov
+        prov = prov.vms
+      else
+        prov = prov::Vm
+      end
+
+      prov.find_by(dits)
+    end
+
+    def find_host_by_bios_uuid(new_result)
+      uuid_provider_types = [ManageIQ::Providers::Redhat::InfraManager::Vm.name,
+                             ManageIQ::Providers::Openstack::CloudManager::Vm.name,
+                             ManageIQ::Providers::Vmware::InfraManager::Vm.name]
+      Vm.find_by(:uid_ems => new_result[:identity_system].downcase,
+                 :type    => uuid_provider_types)
+    end
+
+    def cross_link_node(new_result)
+      # Establish a relationship between this node and the vm it is on (if it is in the system)
+      host_instance = find_host_by_bios_uuid(new_result)
+      if !host_instance && new_result[:identity_infra]
+        host_instance = find_host_instance(new_result[:identity_infra])
+      end
+
+      new_result[:lives_on_id] = host_instance.try(:id)
+      new_result[:lives_on_type] = host_instance.try(:type)
+    end
```

Comment:
@zeari if you enable the cloud provider in kubernetes then `identity_infra` can have a value even on e.g. oVirt.

So this is currently wrong. You have to check what's the provider that is reported by `identity_infra` in order to decide whether to use `find_host_by_provider_id` or not.